core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -562,4 +581,60 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
             // Netty will close the connection automatically on an IOException.
         }
     }
+
+    private void applyOutlierDetection(ClientRequestContext ctx) {
+        if (outlierDetectingRule == null || !isAcquirable()) {
+            return;
+        }
+        assert outlierDetector != null;
+        if (outlierDetector.isOutlier()) {
+            return;
+        }
+
+        ctx.log().whenComplete().thenAccept(this::detectOutlier);
+    }
+
+    private void detectOutlier(RequestLog log) {
+        final RequestContext context = log.context();
+        if (!context.eventLoop().inEventLoop()) {
+            // Execute in the event loop to prevent a connection from being marked as an outlier as a request is
+            // about to be sent.
+            detectOutlier(log);
+            return;
+        }
+
+        final OutlierDetectionDecision decision;
+        try {
+            decision = outlierDetectingRule.decide(context, log.responseHeaders(), log.responseCause());
+        } catch (Exception e) {
+            logger.warn("Unexpected exception from an OutlierDetectingRule: {}", outlierDetectingRule, e);
+            return;
+        }
+
+        if (decision == null) {
+            return;
+        }
+        switch (decision) {
+            // NEXT is assumed as SUCCESS.
+            case SUCCESS:
+            case NEXT:
+                outlierDetector.onSuccess();
+                // The connection was marked as an outlier, but it's back to normal.
```

Comment:
`OutlierDetector` is not a circuit breaker which can be closed after the open state.
Although the last request succeeded luckily, you should see the total result in a window rather than just looking at one result.

Reply:
I didn't realize that most of `OutlierDetector.onSuccess()` call doesn't change the state(`isOutlier`) of the detector before looking into the implementation detail.
Could you also add your explanation to the comment?

Reply:
Done.